pkg/registry/comparison/comparison.go

```diff
@@ -0,0 +1,182 @@
+package comparison
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/caicloud/nirvana/log"
+	"github.com/kleveross/klever-model-registry/pkg/common"
+	"github.com/kleveross/klever-model-registry/pkg/registry/errors"
+	"github.com/kleveross/klever-model-registry/pkg/registry/harbor"
+	"github.com/kleveross/klever-model-registry/pkg/registry/paging"
+	"github.com/kleveross/klever-model-registry/pkg/util"
+	ormbmodel "github.com/kleveross/ormb/pkg/model"
+)
+
+// ComparisonGenerator generator model list and
+// compare model diff
+func ComparisonGenerator(ctx context.Context, models Comparison, opt *paging.ListOption) (*paging.Result, error) {
+	metaList, err := composeComparison(models.Model)
+	if err != nil {
+		return nil, err
+	}
+	return paging.Page(metaList, opt), nil
+}
+
+func composeComparison(models []ComparisonModel) ([]ormbmodel.Model, error) {
+	proxy := harbor.NewProxy(common.ORMBDomain, common.ORMBUserName, common.ORMBPassword)
+	metaList := make([]ormbmodel.Model, 0)
```

Comment:
In following loop, `metaList` have only one element? (becasue have a `break` statement), so `metaList ` is not need slice,  one object is OK or not?

Reply:
`break` only break the tag loop, we need it to be a slice since we'll have different models to be compared.